src/views/CountDetail.vue

```diff
@@ -409,36 +373,78 @@ function selectSearchBarText(event) {
   })
 }
 
-async function searchProducts() {
-  updateFilteredItems(); 
-}
+async function scanProduct() {
+  if(!queryString.value) {
+    showToast(translate("Please provide a valid barcode identifier."))
+    return;
+  }
 
-function updateFilteredItems() {
-  if (!queryString.value.trim()) {
-    filteredItems.value = itemsList.value;
+  const barcodeIdentifier = productStoreSettings.value["barcodeIdentificationPref"];
+  const cachedProducts = getCachedProducts.value;
+
+  let selectedItem = {}
+
+  if(cycleCount.value.statusId === 'INV_COUNT_ASSIGNED') {
```

Comment:
We should think on improving this if-else block here, as we have the same logic in else part which is similar to the if block when selected item is not found and we just have difference of itemStatusId check in overall logic.